model/ftn/wmesmfmd.ftn

```diff
@@ -4626,6 +4933,135 @@
+!
+!     Setup merging mask field for an import field for the cases that
+!     model domains does not overlap completely
+!
+!  2. Method :
+!
+!  3. Parameters :
+!
+!     Parameter list
+!     ----------------------------------------------------------------
+!       mmskField   Type   I/O merging mask field
+!       impField    Type   I   import field
+!       fillVal     Real   I   fill value
+!       mskCreated  Log.   I/O mask is created or not 
+!       rc          Int    O   Return code
+!     ----------------------------------------------------------------
+!
+!  4. Subroutines used :
+!
+!      Name      Type  Module   Description
+!     ----------------------------------------------------------------
+!      NONE
+!     ----------------------------------------------------------------
+!
+!  5. Called by :
+!
+!  6. Error messages :
+!
+!  7. Remarks :
+!
+!  8. Structure :
+!
+!  9. Switches :
+!
+! 10. Source code :
+!
+!/ ------------------------------------------------------------------- /
+!/
+!/ ------------------------------------------------------------------- /
+!/ Parameter list
+!/
+      implicit none
+      type(ESMF_Field)   :: mmskField
+      type(ESMF_Field)   :: impField
+      real(ESMF_KIND_RX) :: fillVal
+      logical            :: mskCreated
+      integer, optional  :: rc
+!/
+!/ ------------------------------------------------------------------- /
+!/ Local parameters
+!/
+      integer, parameter :: lde = 0
+      integer :: i, j
+      integer :: elb(2), eub(2)
+      integer :: tlb(2), tub(2)
+      real(ESMF_KIND_RX), pointer :: mptr(:,:), dptr(:,:)
+      real(ESMF_KIND_RX), pointer :: mmsk(:,:)
+      character(ESMF_MAXSTR) :: fnm
+#if defined(TEST_WMESMFMD) || defined(TEST_WMESMFMD_SETUPIMPMMSK)
+      integer, save :: timeSlice = 1
+#endif
+!
+! -------------------------------------------------------------------- /
+! Check mask is created or not
+!
+      if ( mskCreated ) return
+!
+! -------------------------------------------------------------------- /
+! Set up fields and pointers
+!
+
+      call ESMF_FieldGet( impField, name=fnm, rc=rc )
+      if (ESMF_LogFoundError(rc, PASSTHRU)) return
+
+      if ( impGridIsLocal ) then
+        call ESMF_FieldGet( impMask, localDE=lde, farrayPtr=mptr, &
+          exclusiveLBound=elb, exclusiveUBound=eub, &
+          totalLBound=tlb, totalUBound=tub, rc=rc )
+        if (ESMF_LogFoundError(rc, PASSTHRU)) return
+        call ESMF_FieldGet( impField,  localDE=lde, farrayPtr=dptr, rc=rc )
+        if (ESMF_LogFoundError(rc, PASSTHRU)) return
+        call ESMF_FieldGet( mmskField, localDE=lde, farrayPtr=mmsk, rc=rc )
+        if (ESMF_LogFoundError(rc, PASSTHRU)) return
+      endif
+!
+! -------------------------------------------------------------------- /
+! Create merging mask
+!
+      if ( impGridIsLocal ) then
+        do j = elb(2),eub(2)
+          do i = elb(1),eub(1)
+            mmsk(i,j) = 0.0
+            if (dptr(i,j).lt.fillVal) then
```

Comment:
Using less than catches more possible fill values but do we want to catch more possible fill values? Ideally fillVal = NaN on systems where this is available because a NaN will propagate if used in interpolation.

Reply:
As @BinLiu-NOAA is suggested I tried to use `if (abs(dptr(i,j)-fillVal) < 1.0e-8) then` but this is failing and I think that it is not creating merge mask as expected. I'll also plaining to test equal. @danrosen25 Let me know if you have any other suggestion.

Reply:
BTW, the type of the dptr and fill Value is `ESMF_KIND_RX`which is R4 in WW3 cap by default. In CMEPS the type is R8. So, this could be a source of the problem and explains why @BinLiu-NOAA suggestion is not working. Maybe it is better to keep as its.

Reply:
I tested with equal and it is failing too. So, at this point it is better to keep as it is. We could try to build model with R8 but as I know it is not easy under UFS model and R8 needs to be default for CMEPS coupling of course if it works.

Reply:
@danrosen25 if you don't have any objection about it, I'll resolve this.